### DISCUSSION TITLE
Strongly recommend adding the ability to update core code online

### DISCUSSION THREAD

**Discussion by YHKMCOM** [AdguardTeam/AdGuardHome#6153] [Q&A]
I always want to use the latest official version, but there is no online update function. I have to delete the original program, download the latest version again, and then reinstall and reconfigure it, which is very annoying. I strongly recommend adding the online update core function to the official AdGuardHome, so that we can keep the latest version with the official at any time. Please consider my suggestion. Very much needed.

**Comment by ainar-g** [comment]
There is already an autoupdate feature, if that's what you mean.  Unless you're using Docker, Snap, or some kind of third-party package.

**Reply by YHKMCOM** [reply]
That's right, the docker I used.Can we add an automatic update feature?

**Reply by ainar-g** [reply]
Docker images are themselves versioned, so people using Docker are setting up automatic updates themselves.  Snapcraft, iirc, has automatic updates on by default.

**Reply by YHKMCOM** [reply]
I did not find any automatic updates for the Docker version I am using. It is the official Docker version.

**Reply by luzfcb** [reply]
You probably need to use a tool to check if a new version of a docker image used by the containers exists, pull the new version, and update the containers.

watchtower does exactly this task. Check the https://containrrr.dev/watchtower/

**Comment by AaronIsMyBrother** [comment]
I would like to ask you a question. Did you try to start AG with not 'root' user?
